### PR TITLE
doc: configuration and folder cleanup

### PR DESCRIPTION
**Pull request by gmarull**
Prefix Sphinx related folders (static, scripts, extensions...) with an underscore, same as it was done for `_templates`. This emphasizes that these folders do not contain actual documentation. `_static` content is also sorted into categories (js, css...).

I have also simplified/cleaned the `conf.py` file.

### CODE REVIEW

**Comment by gmarull** [comment]
Looks like compliance check doesn't allow empty `__init__.py` files (were added previous to these changes...). This is a valid case and a frequent situation in Python code. cc: @galak @nashif

**Review by carlescufi — COMMENTED** [review]
Very nice cleanup. I am just not sure about the underscores before the folders. From what you write in the commit messages, I don't see that they are required at all, and we have no precedent in the tree with leading underscores on folder names. I understand the point of making folders that do not contain actual doc more visible, but could we achieve that instead in some other way?

**Comment by carlescufi** [comment]
> Looks like compliance check doesn't allow empty `__init__.py` files (were added previous to these changes...). This is a valid case and a frequent situation in Python code. cc: @galak @nashif

Just add a comment like:
`# File intentionally empty`
inside the file?

**Comment by gmarull** [comment]
> Very nice cleanup. I am just not sure about the underscores before the folders. From what you write in the commit messages, I don't see that they are required at all, and we have no precedent in the tree with leading underscores on folder names. I understand the point of making folders that do not contain actual doc more visible, but could we achieve that instead in some other way?

I'd say it is a fairly common practice in Sphinx. We actually do it for templates (we have `_templates` for Sphinx template customizations and `templates` for some documentation content template). This way there is a clear split between what are docs and Sphinx related files.

**Review by nashif — APPROVED** [review]

File: doc/_extensions/lexer/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2021 Zephyr Project members and individual contributors
```
Comment (mbolivar-nordic) [inline doc/_extensions/lexer/__init__.py:1]: Not Nordic?

**Review by mbolivar-nordic — APPROVED** [review]
Great cleanup!

File: doc/_extensions/lexer/__init__.py (hunk 1 above)
Comment (gmarull) [reply doc/_extensions/lexer/__init__.py:1]: These are just comments to pass compliance checks. As the files were already there I thought it was better to add this copyright message.

Comment (nashif) [reply doc/_extensions/lexer/__init__.py:1]: i guess nobody wants to own copyright on an empty file :-D

**Comment by carlescufi** [comment]
> > Very nice cleanup. I am just not sure about the underscores before the folders. From what you write in the commit messages, I don't see that they are required at all, and we have no precedent in the tree with leading underscores on folder names. I understand the point of making folders that do not contain actual doc more visible, but could we achieve that instead in some other way?
> 
> I'd say it is a fairly common practice in Sphinx. We actually do it for templates (we have `_templates` for Sphinx template customizations and `templates` for some documentation content template). This way there is a clear split between what are docs and Sphinx related files.

Fair enough.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**